### DISCUSSION TITLE
[@types/webextension-polyfill] Events.Event<T> Issue

### DISCUSSION THREAD

**Discussion by vozeldr** [DefinitelyTyped/DefinitelyTyped#56665] [Issues with a @types package]
I've started having a problem in WebStorm using webextension-polyfill. As an example:

```typescript
const port = browser.runtime.connectNative('myapp');
port.onMessage.addListener((message) => {});
```

The addListener correctly appears with the proper signature when I type the `port.onMessage.` but upon completing `addListener` it is getting picked up as being the method:

```
deprecated MediaQueryList.addListener(callback: ((this:MediaQueryList, ev: MediaQueryListEvent) => any) | null):    void
```

This means I lose the typing of the callback's parameter too because it believes it to be `MediaQueryListEvent`.

I'm honestly not sure if the problem lies in WebStorm or TypeScript or in the type definition in `events.d.ts`. I did file an [issue in YouTrack](https://youtrack.jetbrains.com/issue/WEB-53211) also.

[a comment by an automated account is not shown]

**Comment by vozeldr** [comment]
The problem was due to WebStorm... something in my .idea folder was causing the problem.

**Comment by Lusito** [comment]
Just to be clear, you found a solution? If you have more details, it might be worth sharing. I'm not using WebStorm, so I can't reproduce this.
